# Exception::__construct




<div class="phpcode"><span class="html">
For those that haven&apos;t done exception chaining. Here&apos;s an example. <br><br>This allows you to add the previous exception to the next one and give yourself detailed information in the end as to what happened. This is useful in larger applications.<br><br><span class="default">&lt;?php<br></span><span class="keyword">function </span><span class="default">theDatabaseObj</span><span class="keyword">(){<br>&#xA0; &#xA0;&#xA0; if( </span><span class="default">database_object </span><span class="keyword">){<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; return </span><span class="default">database_object</span><span class="keyword">; <br>&#xA0; &#xA0;&#xA0; }<br>&#xA0; &#xA0;&#xA0; else{<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; throw new </span><span class="default">DatabaseException</span><span class="keyword">(</span><span class="string">&quot;Could not connect to the database&quot;</span><span class="keyword">);<br>&#xA0; &#xA0;&#xA0; }<br>}<br><br>function </span><span class="default">updateProfile</span><span class="keyword">( </span><span class="default">$userInfo </span><span class="keyword">){<br>&#xA0; &#xA0;&#xA0; try{<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; </span><span class="default">$db </span><span class="keyword">= </span><span class="default">theDatabaseObj</span><span class="keyword">();<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; </span><span class="default">$db</span><span class="keyword">-&gt;</span><span class="default">updateProfile</span><span class="keyword">();<br>&#xA0; &#xA0;&#xA0; }<br>&#xA0; &#xA0;&#xA0; catch( </span><span class="default">DatabaseException $e </span><span class="keyword">){<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; </span><span class="default">$message </span><span class="keyword">= </span><span class="string">&quot;The user :&quot; </span><span class="keyword">. </span><span class="default">$userInfo</span><span class="keyword">-&gt;</span><span class="default">username </span><span class="keyword">. </span><span class="string">&quot; could not update his profile information&quot;</span><span class="keyword">;<br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; </span><span class="comment">/* notice the &apos;$e&apos;. I&apos;m adding the previous exception&#xA0; to this exception. I can later get a detailed view of <br>&#xA0; &#xA0; &#xA0; &#xA0; &#xA0; where the problem began. Lastly, the number &apos;12&apos; is&#xA0; an exception code. I can use this for categorizing my <br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; exceptions or don&apos;t use it at all. */ <br>&#xA0; &#xA0; &#xA0; &#xA0;&#xA0; </span><span class="keyword">throw new </span><span class="default">MemberSettingsException</span><span class="keyword">(</span><span class="default">$message</span><span class="keyword">,</span><span class="default">12</span><span class="keyword">,</span><span class="default">$e</span><span class="keyword">);<br>&#xA0; &#xA0;&#xA0; }<br>}<br><br>try{<br>&#xA0; &#xA0;&#xA0; </span><span class="default">updateProfile</span><span class="keyword">( </span><span class="default">$userInfo </span><span class="keyword">);<br>}<br>catch( </span><span class="default">MemberSettingsException $e </span><span class="keyword">){<br>&#xA0; &#xA0;&#xA0; </span><span class="comment">// this will give all information we have collected above. <br>&#xA0; &#xA0;&#xA0; </span><span class="keyword">echo </span><span class="default">$e</span><span class="keyword">-&gt;</span><span class="default">getTraceAsString</span><span class="keyword">();<br>}<br></span><span class="default">?&gt;</span>
</span>
</div>
  

#

[Official documentation page](https://www.php.net/manual/en/exception.construct.php)

**[â¬† to root](/)**